python-package/lightgbm/basic.py

```diff
@@ -2186,6 +2187,8 @@ def _lazy_init(
         elif isinstance(data, np.ndarray):
             self.__init_from_np2d(data, params_str, ref_dataset)
         elif _is_pyarrow_table(data):
+            if not CFFI_INSTALLED:
+                raise LightGBMError("Cannot init dataframe from Arrow without `pyarrow` and `cffi` installed.")
```

Comment:
```suggestion
```

I think this one is unnecessary... the very next statement is a call to `self.__init_from_pyarrow_table()`, which will raise exactly this same error.

Could you please remove this, or if you disagree explain why this should be here?